core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -296,6 +296,14 @@ abstract class RawModule extends BaseModule {
   protected[chisel3] override def moduleBuilt(): Unit = _afterModuleBuilt.foreach(_())
 
   private[chisel3] def initializeInParent(): Unit = {}
+
+  /** Record the layers that are known to this [[RawModule]] at the time it was
+    * created.
+    *
+    * This is needed if this [[RawModule]] is ever converted to a [[BlackBox]]
+    * or [[ExtModule]].
+    */
+  final private[chisel3] def knownLayers: Seq[layer.Layer] = Builder.layers.toSeq
```

Comment:
This will be re-evaluated on every use, right?

Reply:
I think that's probably OK. I was worried we were stashing the layers BEFORE running the module's body.